builder/builder/builder.py

```diff
@@ -241,16 +242,218 @@ def ConstructSnakefileConfig(self) -> dict:
+        """Packages a remote module into the build directory
+
+        Assumes node.snakemake is a dict with the following structure:
+        {
+            "function": "github",
+            "args": ["repo/name"],
+            "kwargs": {
+                "branch": "main",  # key can also be "tag" or "commit"
+                "path": "workflow/Snakefile"
+            }
+        }
+        """
+        # Validate node.snakemake
+        if not isinstance(node.snakefile, dict):
+            raise ValueError(
+                "Remote module configuration expected, got: " + str(node.snakefile)
+            )
+        if node.snakefile["function"] not in ["github"]:
+            raise ValueError(
+                "Only github function is currently supported, got: "
+                + node.snakefile["function"]
+            )
+        if not any(
+            [k in node.snakefile["kwargs"] for k in ["branch", "tag", "commit"]]
+        ):
+            raise ValueError(
+                "Remote module requires a branch, tag or commit to be specified, kwargs: "
+                + str(node.snakefile["kwargs"])
+            )
+        if "path" not in node.snakefile["kwargs"]:
+            raise ValueError(
+                "Remote module requires a path to be specified, kwargs: "
+                + str(node.snakefile["kwargs"])
+            )
+        # Identify remote folder structure
+        try:
+            m_path = node.snakefile["kwargs"]["path"]
+            m_pathlist = m_path.split(os.path.sep)[:-1]
+            m_workflow_foldername = m_pathlist[-1]
+            m_modulename_foldername = m_pathlist[-2]
+            m_type_foldername = m_pathlist[-3]
+            m_project_foldername = m_pathlist[-4]
+            m_workflows_foldername = m_pathlist[-5]
+            if m_workflows_foldername != "workflows":
+                raise IndexError
+            m_repo_name = os.path.join(*node.snakefile["args"][0].split("/"))
+        except IndexError:
+            raise ValueError("Module Snakefile is not in the expected folder structure")
+        # Recreate folder structure in the build directory
+        dest = os.path.join(
+            build_path,
+            "workflow",  # base 'workflow' folder
+            "modules",  # downloaded modules are stored in a 'modules' sub-folder
+            m_repo_name,
+        )
+        pathlib.Path(dest).mkdir(parents=True, exist_ok=True)
+        branch = node.snakefile["kwargs"].get("branch", None)
+        if not branch:
+            branch = node.snakefile["kwargs"].get("tag", None)
+        if not branch:
+            branch = node.snakefile["kwargs"].get("commit", None)
+        if not branch:
+            raise ValueError(
+                "Remote module requires a branch, tag or commit to be specified, kwargs: "
+                + str(node.snakefile["kwargs"])
+            )
+        # Copy github directory structure to the build directory
+        blobs = self.GetRemoteModule_BlobTree(node)
```

Comment:
Some of these functions seem to be static, worth putting a `@staticmethod` decorator on them